client/post-editor/editor-gutenberg-opt-in-notice/index.jsx

```diff
@@ -33,13 +33,14 @@ class EditorGutenbergOptInNotice extends Component {
 		showDialog: PropTypes.func,
 		optInEnabled: PropTypes.bool,
 		noticeDismissed: PropTypes.bool,
+		sidebarOpen: PropTypes.bool,
 		dismissNotice: PropTypes.func,
 	};
 
 	dismissNotice = () => this.props.dismissNotice( 'gutenberg_nudge_notice_dismissed', true );
 
 	render() {
-		if ( ! this.props.optInEnabled || this.props.noticeDismissed ) {
+		if ( ! this.props.optInEnabled || this.props.noticeDismissed || this.props.sidebarOpen ) {
```

Comment:
if we want to easy avoid the awkwardness of repeatedly opening and closing the sidebar we can move from checking `sidebarOpen` to `sidebarHasBeenOpen`

```js
getDerivedStateFromProps( props, state ) {
	if ( ! state.hasOpenedSidebar && props.sidebarOpen ) {
		return { hasOpenedSidebar: true };
	}
}
```

this is a "latch" and will hold open the first time the sidebar is opened, that way once the banner disappears it will stay hidden on account of the sidebar

```js
if ( ! this.props.optInEnabled || this.props.noticeDismissed || this.state.hasOpenedSidebar ) {
```

Reply:
Oh awesome, thank you – great idea. Works wonderfully.